firestore/FirestoreExampleUITests/FirestoreUITest.m

```diff
@@ -28,24 +28,28 @@
 @interface FirestoreUITest : XCTestCase
 @end
 
-@implementation FirestoreUITest {
-  XCUIApplication *_app;
-}
+static XCUIApplication *sApp;
 
-- (void)setUp {
-  [super setUp];
-  _app = [[XCUIApplication alloc] init];
-  [_app launch];
+@implementation FirestoreUITest
+
++ (void)setUp {
+  sApp = [[XCUIApplication alloc] init];
```

Comment:
Im curious, what is the difference here and why does that change the behavior? Before it was an ivar (correct?) and now it's a `static` what? (it's not a property right?)

Reply:
The key change is moving the signin flow from the setup instance method that runs for every test to the setup class method that runs once for all the tests.

An alternative fix would have been to add an `@synchronized(self)` around the signin check and call.

Reply:
Ahh, makes sense now, thanks. I did not notice the transition from instance to class method 😅